File: flutter/lib/domain/models/sync/sync_data.dart
enum DataSource {
  unsynced,
  synced;

  String get locKey => switch (this) {
    .unsynced => 'dataSource.unsynced',
    .synced => 'dataSource.synced',
  };
}

class SyncData<T> {
  final T data;
  final DataSource source;

  SyncData({required this.data, required this.source});
}

File: cmp/composeApp/src/commonMain/kotlin/org/example/crossplatformshowcase/kmp/domain/entities/SyncData.kt
package org.example.crossplatformshowcase.kmp.domain.entities

enum class DataSource {
    UNSYNCED, SYNCED;

    val locKey: String
        get() = when (this) {
            UNSYNCED -> "dataSource.unsynced"
            SYNCED -> "dataSource.synced"
        }
}

data class SyncData<T>(
    val data: T,
    val source: DataSource,
)
